src/engine/strat_engine/blockdev.rs

```diff
@@ -129,10 +135,40 @@ impl BlockDev for StratBlockDev {
     fn devnode(&self) -> PathBuf {
         self.devnode.clone()
     }
+
+    fn user_id(&self) -> &Option<String> {
+        &self.location
+    }
+
+    fn set_user_id(&mut self, location: Option<&str>) -> EngineResult<()> {
+        self.location = location.map(|x| x.to_owned());
+        Ok(())
+    }
+
+    fn hardware_id(&self) -> &Option<String> {
+        &self.disk_id
+    }
+
+    fn initialization_time(&self) -> DateTime<Utc> {
+        Utc.timestamp(self.bda.initialization_time() as i64, 0)
+    }
+
+    fn total_size(&self) -> Sectors {
+        self.current_capacity()
```

Comment:
Whoops! I think what this really ought to be is:

```
self.avail_range().1
```

Reply:
ok